stats/stats_test.go

```diff
@@ -940,17 +940,21 @@ func TestClientStatsUnaryRPCError(t *testing.T) {
 		failfast:   failfast,
 	}
 
-	checkFuncs := []func(t *testing.T, d *gotData, e *expectedData){
-		checkBegin,
-		checkOutHeader,
-		checkOutPayload,
-		checkInHeader,
-		checkInTrailer,
-		checkEnd,
+	checkFuncs := map[int]*checkFuncWithCount{
```

Comment:
It's not guaranteed that the keys are consecutive. If I use a list, I need to check if the value is nil.